From 464c651d49989c154c81bf869b4f3388d096d789 Mon Sep 17 00:00:00 2001
From: Brendan O'Dea <bod@debian.org>
Date: Fri, 16 Dec 2005 01:32:14 +1100
Subject: Downgrade the optimization of sv.c on arm due to a gcc-4.4 bug

Bug-Debian: http://bugs.debian.org/580334

Regression from gcc-4.3, not reported yet.

 perl -e '"-2" =~ /(.+)/; @foo=(); push @foo, $_ for $1..undef; print @foo ? "ok\n" : "not ok\n"'


Patch-Name: debian/arm_optim.diff
---
 cflags.SH |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/cflags.SH b/cflags.SH
index f57cf2f..54574b2 100755
--- a/cflags.SH
+++ b/cflags.SH
@@ -313,7 +313,11 @@ for file do
     regexec) ;;
     run) ;;
     scope) ;;
-    sv) ;;
+    sv)
+	case $archname in
+	arm-*|armeb-*)
+	    optimize=-O0;;
+	esac;;
     taint) ;;
     toke) ;;
     universal) ;;
-- 
1.7.5.1

